### PR TITLE
Add working directory global parameter

### PR DESCRIPTION
**Pull request by pellared**
## Why

> Why this pull request was created? Explain the problem that this pull request is trying to solve.

Fix https://github.com/goyek/goyek/issues/69

## What

> What does this pull request contain? Explain what is this pull request changing.

- Merge `param` and `paramValueFactory` types into `registeredParam` as they felt redundant
- Add `wd` global parameter allowing to change the working directory.
- Reuse `wd` parameter in wrapper scripts.
- Cleanup examples and build pipeline.

Refactoring proposals are welcome. I was thinking of introducing the "global parameter" abstraction, but could not figure out any obvious implementation in a dozen minutes. I felt that the current code is probably "maintainable" enough.

Test coverage is a little lower, because of panics which are probably almost impossible to test.

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated.
- [ ] The code changes follow [Effective Go](https://golang.org/doc/effective_go).

### CODE REVIEW

File: taskflow_test.go [hunk 1]
```diff
@@ -508,7 +508,8 @@ func Test_wd_param(t *testing.T) {
 	flow := &goyek.Taskflow{}
 	beforeDir, err := os.Getwd()
 	requireEqual(t, err, nil, "should get work dir before the taskflow")
-	dir := tempDir(t)
+	dir, cleanup := tempDir(t)
```
Comment (breml) [inline taskflow_test.go:511]: Why don't you use https://golang.org/pkg/testing/#T.TempDir.
In this case, `go test` takes care of the cleanup.

Comment (pellared) [reply taskflow_test.go:511]: I did use it initially but the backwards compatibility builds started failing. The method was introduced in Go 1.15. I prefer to keep the backwards compatibility as long as it is not expensive to maintain. Which is not so far.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Add Bash and PowerShell wrapper scripts.
+- Add `-wd` global parameter allowing to change the working directory.
```
Comment (agabrys) [inline CHANGELOG.md:13]: Consider add information about the new method: `Taskflow.WorkDirParam`.

The following line were added for `Taskflow.VerboseParam`:
> - Remove `TF.Verbose`, add `Taskflow.VerboseParam` instead.

Comment (pellared) [reply CHANGELOG.md:13]: Tried to address 4f7ab5b40abd41ad5d433fce60f216ab542c3134

**Review by dertseha — APPROVED** [review]

**Pull request merged**